app/models/event.rb

```diff
@@ -94,4 +99,20 @@ def self.group_by_month(year_events, month, year)
     end
   end
 
+  def self.find_current_event_id
+    sql = %{
+      start_date <= ? AND
+      end_date >= ?
+    }
+    current_event = Event.all.where(sql, Date.today, Date.today).order(start_date: :asc).first.id rescue nil
+    return current_event if current_event.present?
+
+    # We haven't found the current_event so we need to find the next event
+    sql = %{
+      start_date > ?
+    }
+
+    next_event = Event.all.where(sql, Date.today).order(start_date: :asc).first.id rescue nil
+  end
```

Comment:
I think this whole function can be simplified with just one query now.
Since we need to retrieve just one event, it should be sufficient to get the closest events to the current date like so: `where("start_date >= ?", Date.today)`, and then pick the first one after ordering the retrieved events.

Also, you don't need `Event.all.where`, but `Event.where` is enough

Reply:
Actually, forget what I just said about the first point, we need to consider the `end_date` as well :) So you can just change the `Event.all` thing for now

Reply:
Thanks @Levia I have made that change.